junit-platform-launcher/src/main/java/org/junit/platform/launcher/core/LauncherDiscoveryRequestBuilder.java

```diff
@@ -268,14 +273,24 @@ private LauncherDiscoveryListener getLauncherDiscoveryListener(ConfigurationPara
 				.map(value -> LauncherDiscoveryListeners.fromConfigurationParameter(
 					DEFAULT_DISCOVERY_LISTENER_CONFIGURATION_PROPERTY_NAME, value)) //
 				.orElseGet(LauncherDiscoveryListeners::abortOnFailure);
-		if (discoveryListeners.isEmpty()) {
+
+		List<LauncherDiscoveryListener> serviceLoaderListeners = configurationParameters.getBoolean(
+			DISCOVERY_LISTENERS_AUTODETECTION_ENABLED_PROPERTY_NAME).filter(p -> p).map(
+				p -> StreamSupport.stream(
+					new ServiceLoaderLauncherDiscoveryListenerRegistry().loadListeners().spliterator(), false).collect(
+						Collectors.<LauncherDiscoveryListener> toList())).orElse(Collections.emptyList());
```

Comment:
Let's try to trick Spotless into formatting this in a more readable way. Sth. like this should work:
```suggestion
		List<LauncherDiscoveryListener> serviceLoaderListeners = configurationParameters.getBoolean(
			DISCOVERY_LISTENERS_AUTODETECTION_ENABLED_PROPERTY_NAME) //
				.filter(p -> p) //
				.map(p -> StreamSupport.stream(new ServiceLoaderLauncherDiscoveryListenerRegistry().loadListeners().spliterator(), false) //
					.collect(Collectors.<LauncherDiscoveryListener>toList())) //
				.orElse(Collections.emptyList());
```